[KNI] rewrite the commit-msg hook

ffromani opened the pull request:
Rewrite the script that verifies the commit message in go, taking into
consideration the project rules described in RESYNC.README.md.

ffromani committed:
[KNI] ci: rewrite the commit-msg hook

Rewrite the script that verifies the commit message in go, taking into
consideration the project rules described in RESYNC.README.md.

Signed-off-by: Shereen Haj <shajmakh@redhat.com>

shajmakh committed:
[KNI] ci: move verify-commit logic entirely to go tool

Now that we have a tool that verifies a single commit message, enhance
it even more to deal with N commits and do the rest of the
verification logic.
From a real example, it turns out that the Signed-off line is prefixed
with spaces, thus also fix the isKonflux method to deal with this input.

Assisted-by: Cursor v1.2.2
AI-Attribution: AIA HAb Hin R Cursor v1.2.2 v1.0
Signed-off-by: Shereen Haj <shajmakh@redhat.com>

shajmakh commented:
check here to see new verify-commit CI logs:
https://github.com/openshift-kni/scheduler-plugins/actions/runs/18033941878/job/51316471387?pr=418#:~:text=9-,2025/09/26%2009%3A35%3A34%20using%20branch%3A%20HEAD,2025/09/26%2009%3A35%3A34%20commit%20message%20has%201%20lines,-Post%20Set%20up

shajmakh commented:
@ffromani this should be ready for review

shajmakh committed:
[KNI] ci: bump actions versions

Update to the latest for checkout and go setup.
https://github.com/actions/setup-go/tags
https://github.com/actions/checkout/tags

Signed-off-by: Shereen Haj <shajmakh@redhat.com>

shajmakh committed:
[KNI] ci: run commit verification on all product branches

Isolate the commits verification and support running it on all product
branches.

Assisted-by: Cursor v1.2.2
AI-Attribution: AIA Primarily human, Content edits, Human-initiated, Reviewed, cursor v1.2.2 v1.0

Signed-off-by: Shereen Haj <shajmakh@redhat.com>

shajmakh committed:
[KNI] ci: run verify-commit tool tests

To ensure nothing gets merged broken.

Signed-off-by: Shereen Haj <shajmakh@redhat.com>
